### PR TITLE
Use posix-spawn instead of Open3

### PR DESCRIPTION
**Pull request by RobWu**
posix-spawn (https://github.com/rtomayko/posix-spawn)

Use `POSIX::Spawn::Child.new` instead of `Open3.popen3`

See also 

> Q: Why use POSIX::Spawn::Child instead of popen3, hand rolled fork/exec code, or Process::spawn?

https://github.com/rtomayko/posix-spawn/blob/master/lib/posix/spawn/child.rb#L44

### CODE REVIEW

**Comment by RobWu** [comment]
Looks like 'posix-spawn' don't support ruby

File: lib/diffy/diff.rb [hunk 1]
```diff
@@ -37,7 +37,8 @@ def initialize(string1, string2, options = {})
       if ! ['strings', 'files'].include?(@options[:source])
         raise ArgumentError, "Invalid :source option #{@options[:source].inspect}. Supported options are 'strings' and 'files'."
       end
-      @string1, @string2 = string1, string2
+      @string1 = string1
+      @string2 = string2
```
Comment (NARKOZ) [inline lib/diffy/diff.rb]: why did you change this?
